src/compiler/moduleSpecifiers.ts

```diff
@@ -251,7 +239,10 @@ namespace ts.moduleSpecifiers {
             let pathsInDirectory: string[] | undefined;
             allFileNames.forEach((canonicalFileName, fileName) => {
                 if (startsWith(canonicalFileName, directoryStart)) {
-                    (pathsInDirectory || (pathsInDirectory = [])).push(fileName);
+                    // If the importedFile is from node modules, use only paths in node_modules folder as option
+                    if (!importedFileFromNodeModules || pathContainsNodeModules(fileName)) {
+                        (pathsInDirectory || (pathsInDirectory = [])).push(fileName);
+                    }

```

Comment:
I see the explanation of this change in the PR description. It looks like everything outside this file is just cleanup/refactoring or test changes—is that right, or is there somewhere else I should look for behavior changes?

Reply:
Thats correct.. The real change is two changes in this file (casing and node_modules behaviour) everything else was just refactoring for cleaner interface and potential new additions needed from program..